### ISSUE TITLE
Array.isArray() type guard doesn't recognize ReadonlyArray<T>

### ISSUE THREAD

**Issue by jrockwood** [Microsoft/TypeScript#19892]
**TypeScript Version:**  2.6.1

**Code**

```ts
interface StringRecord {
    readonly message: string | ReadonlyArray<string>; 
}

const record: StringRecord = { message: 'item' };
const message: string = Array.isArray(record.message) ? record.message.join('') : record.message;
```

**Expected behavior:**
The `Array.isArray()` type guard check should detect that `record.message` is a string. Note that if the `message` field is changed to `string | string[]` then everything works as expected. It's only when it's a `ReadonlyArray<T>` that it fails to compile. I can definitely add a `as string` cast, but that's kind of ugly and potentially dangerous if I change the data type later.

**Actual behavior:**
An error is reported: `error TS2322: Type 'string | ReadonlyArray<string>' is not assignable to type 'string'.`

**Comment by andy-ms** [comment]
Looks like a duplicate of #17002.
